liquibase-core/src/main/java/liquibase/resource/PathHandlerFactory.java

```diff
@@ -61,48 +62,37 @@ public OutputStream createResource(String resourcePath) throws IOException {
 
     /**
      * Return the resource for the given path.
-     * @param fallbackToResourceAccessor if true, check {@link Scope#getResourceAccessor()} before returning null
-     * @return null if the resource does not exist
      *
+     * @param includeResourceAccessor if true, check {@link Scope#getResourceAccessor()} as well.
+     * @return null if the resource does not exist
      * @throws IOException if the path cannot be understood or if there is a problem parsing the path
+     * @throws IOException if the path exists as both a direct resourcePath and also in the resourceAccessor (if included). Unless {@link liquibase.GlobalConfiguration#DUPLICATE_FILE_MODE} overrides that behavior.
      */
-    public Resource getResource(String resourcePath, boolean fallbackToResourceAccessor) throws IOException {
-        String searchPath = GlobalConfiguration.SEARCH_PATH.getCurrentValue();
-        PathHandler plugin = determinePlugin(searchPath, resourcePath);
+    public Resource getResource(String resourcePath, boolean includeResourceAccessor) throws IOException {
+        final PathHandler plugin = getPlugin(resourcePath);
```

Comment:
I think these changes will break the case that we had last week.

I also think you might be missing the context of that change from last week. Here's the short version of it: https://datical.atlassian.net/browse/DAT-10443?focusedCommentId=114604

So, with your proposed change here, if I pass a non-absolute path into this method, `getPlugin` is going to give me a `DirectoryPathHandler` which is going to find the file I'm looking for locally, rather than finding it in S3 (despite setting my search path to S3). -- At least, that's what we're surmising. We haven't actually tested this to confirm it (cc @wwillard7800 )

Reply:
If you don't want the duplicate file message then you have to set duplicatefilemode=warn then it will still give the message but will find the first one